substrate/frame/assets-freezer/src/impls.rs

```diff
@@ -0,0 +1,164 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+
+use pallet_assets::FrozenBalance;
+use frame_support::traits::fungibles::{Inspect, InspectFreeze, MutateFreeze};
+use sp_runtime::traits::{Get, Zero};
+
+impl<T: Config<I>, I: 'static> FrozenBalance<AssetIdOf<T, I>, AccountIdOf<T>, AssetBalanceOf<T, I>>
+	for Pallet<T, I>
+{
+	fn frozen_balance(
+		asset: AssetIdOf<T, I>,
+		who: &AccountIdOf<T>,
+	) -> Option<AssetBalanceOf<T, I>> {
+		FrozenBalances::<T, I>::get(asset, who)
+	}
+
+	fn died(asset: AssetIdOf<T, I>, who: &AccountIdOf<T>) {
+		FrozenBalances::<T, I>::remove(asset.clone(), who);
+		Freezes::<T, I>::remove(asset, who);
+	}
+}
+
+// Implement fungibles::Inspect as it is required. To do so, we'll re-export
```

Comment:
`pallet-assets` already implements `fungibles::Inspect`, so it's a bit annoying to have to implement it here again.. I wonder if that's feasible?

```suggestion
// Implement [`fungibles::Inspect`] as it is required. To do so, we'll re-export
```

Reply:
I thought about that too. However, trait dependencies are Pallet-based, not Config-based. So, if `pallet-assets-freezer` wanted to implement `InspectFreeze` and `MutateFreeze`, it had to also implement `Inspect`.

Luckily, as you said, `pallet-assets` already implements it. As this pallet is being described as an extension of the latter, this also means the implementation of the required trait is also quite straightforward.

This `impl` I did was the simplest that could be achieved, at least on the current state of Rust.